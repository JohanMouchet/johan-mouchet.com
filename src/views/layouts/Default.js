// @flow

import * as React from "react";
import { MastHeader, MastFooter } from "views/partials";
import "assets/styles/_styles.scss";

const Default = props => {
  const { children } = props;

  return (
    <>
      <MastHeader />
      {children}
      <MastFooter />
    </>
  );
};

export default Default;
