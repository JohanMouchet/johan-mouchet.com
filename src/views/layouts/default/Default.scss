@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";

.l-layout {
  min-height: 100vh;
  max-width: 1920px;
  display: flex;
  margin: 0 auto;
  padding: t-vertical-rhythm(0.5) t-vertical-rhythm(0.25);

  @media (min-width: map-get($s-breakpoints, sm)) {
    padding: t-vertical-rhythm(0.5) t-vertical-rhythm(1);
  }

  @media (min-width: map-get($s-breakpoints, md)) {
    padding: t-vertical-rhythm(1) t-vertical-rhythm(2);
  }

  &__background {
    position: fixed;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;

    &::before,
    &::after {
      content: "";
      position: absolute;
      top: 0;
      left: 0;
      height: 100%;
      width: 100%;
    }

    &::before {
      background: url("#{map-get($s-paths, theme)}/layout-background.jpg")
        $s-color-primary;
      filter: blur(6px);
      transform: scale(3.5);
    }

    &::after {
      background: url("#{map-get($s-paths, theme)}/layout-grain.png");
    }
  }

  &__inner {
    display: flex;
    flex-direction: column;
    position: relative;
    overflow: hidden;
    width: 100%;
    box-shadow: 0 10px 15px -3px rgba($s-color-black, 0.4),
      0 4px 6px -2px rgba($s-color-black, 0.4);
    background-color: $s-color-white;
    border-radius: 8px;
  }
}
