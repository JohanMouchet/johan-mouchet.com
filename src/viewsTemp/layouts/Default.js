// @flow

import * as React from "react";
import { HTML } from "../partials/HTML";
import { MastHeader } from "../partials/MastHeader";
import { MastFooter } from "../partials/MastFooter";

const ___ = children => {
  return (
    <HTML>
      <MastHeader />
      {children}
      <MastFooter />
    </HTML>
  );
};
