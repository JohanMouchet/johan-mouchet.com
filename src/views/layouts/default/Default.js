// @flow

import * as React from "react";
import { MastHeader, MastFooter } from "views/partials";
import "assets/styles/_styles.scss";
import "./Default.scss";

const Default = ({ children }: Node) => (
  <div className="l-layout">
    <div className="l-layout__background"></div>
    <div className="l-layout__inner">
      <MastHeader />
      {children}
      <MastFooter />
    </div>
  </div>
);

export default Default;
