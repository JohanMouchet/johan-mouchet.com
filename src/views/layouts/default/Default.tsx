import * as React from "react";
import { Header, Footer } from "views/partials";
import "./Default.scss";

const Default = ({ children }: Node) => (
  <div className="l-layout">
    <div className="l-layout__background"></div>
    <div className="l-layout__inner">
      <Header />
      {children}
      <Footer />
    </div>
  </div>
);

export default Default;
