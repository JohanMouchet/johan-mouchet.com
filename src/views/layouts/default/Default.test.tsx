import { render } from "@testing-library/react";
import { Default } from "./Default";

it("Renders <Default /> without crashing", () => {
  expect(
    render(<Default>children</Default>).container.firstChild
  ).toMatchSnapshot();
});
