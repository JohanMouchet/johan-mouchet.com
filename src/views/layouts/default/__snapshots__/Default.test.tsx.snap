// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Default /> without crashing 1`] = `
<div
  class="l-layout"
>
  <header
    class="p-header"
  >
    <h1
      class="p-header__heading"
    >
      Johan
       
      Mouchet
    </h1>
    <span
      class="p-header__tagline"
    >
      Front-End Engineer, Design System Architect
    </span>
  </header>
  children
  <hr />
  <footer
    class="p-footer"
  >
    <div
      class="grid grid-y-center grid-no-wrap"
    >
      <div
        class="cell cell-shrink"
      >
        <span
          class="p-footer__version"
        >
          7.0.0
        </span>
      </div>
      <div
        class="cell cell-grow"
      >
        <p
          class="p-footer__copyright"
        >
          2024
           © 
          Johan MOUCHET
           
          <span
            class="p-footer__copyright-separator"
          >
            —
          </span>
          <br />
           Made in
           
          <span
            class="p-footer__cocorico"
            tabindex="0"
          >
            France
          </span>
        </p>
      </div>
      <div
        class="cell cell-shrink"
      >
        <a
          aria-label="Get to the top"
          class="p-footer__page-top"
          href="#page-top"
        >
          <svg />
        </a>
      </div>
    </div>
  </footer>
</div>
`;
